### PR TITLE
Improve GitHub Actions configuration

### PR DESCRIPTION
**Pull request by anitagraham**
Updates GitHub actions to run on both `push` and `pull_request` events.

This also speeds up all of the actions by only including support for
the various databases if that is what is being tested.

### CODE REVIEW

**Comment by anitagraham** [comment]
This PR should get Refinerycms building again.

P.S. Could have sworn - in fact did swear - that I had squashed all the little commits together. But here they all are.

**Comment by parndt** [comment]
strange, it didn't run any checks..

**Comment by anitagraham** [comment]
Yes. Strange. I have no explanation. How do we force it to run the checks?

**Comment by parndt** [comment]
🤔

File: core/refinerycms-core.gemspec [hunk 1]
```diff
@@ -3,7 +3,7 @@
 require File.expand_path('../core/lib/refinery/version', __dir__)
 
 version = Refinery::Version.to_s
-rails_version = ['>= 6.0.0', '< 7']
+rails_version = ['>= 6.0.0', '<7']
```
Comment (parndt) [inline core/refinerycms-core.gemspec:6]:
```suggestion
rails_version = ['>= 6.0.0', '< 7']
```

**Comment by parndt** [comment]
There we go @anitagraham

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -10,44 +14,38 @@ jobs:
...
+        run: |
+          sudo apt-get install libmysqlclient-dev -y
+          sudo systemctl start mysql.service
+
+      - name: "Set up PostgreSQL using VM's server"
+        if: ${{ env.DB == 'postgresql' }}
+        run: |
+          sudo apt-get install libpq-dev -y
+          sudo systemctl start postgresql.service
+          sudo -u postgres psql -c "CREATE USER postgres WITH SUPERUSER PASSWORD 'postgres'"
```
Comment (parndt) [inline .github/workflows/main.yml:48]:
```suggestion
```

**Comment by parndt** [comment]
@anitagraham a lot of this was handled by #3501 so I've hijacked it to make actions even better + faster. Thanks!

**Pull request merged**